src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java

```diff
@@ -213,6 +210,222 @@ private static void regionMatchesInternal(
     }
   }
 
+  // For standard Kotlin packages, which are named according to the pattern kotlin.*, we append a
+  // whitespace to the package name of the target class so that they are not mangled due to shading.
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.jvm.internal.Intrinsics ",
+      targetMethod = "areEqual")
+  @MethodHook(
+      type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ", targetMethod = "equals")
+  @MethodHook(type = HookType.AFTER, targetClassName = "kotlin.text.StringsKt ",
+      targetMethod = "equals$default")
+  public static void
+  equalsKt(MethodHandle method, Object alwaysNull, Object[] arguments, int hookId,
+      Boolean equalStrings) {
+    if (arguments[0] instanceof String && arguments[1] instanceof String && !equalStrings) {
```

Comment:
Not sure whether we have been doing this consistently with existing hooks, but we should probably also check the size of `arguments` in these hooks.

Reply:
We haven't! I've gone through the CMP hooks where we don't specify a method descriptor and made sure that we check the size of `arguments` before accessing the array.

Reply:
Thanks! They all look safe now.